### ISSUE TITLE
Directory does not exist coldbox-legacy-app-demo/HTMLHelper/models 

### ISSUE THREAD

**Issue by bardware** [bdw429s/coldbox-legacy-app-demo#4]
I called `server start cfengine=adobe@2018.0.01+311402` when I wanted to look at this sample app and was greeted by an error 

``` 
Directory does not exist
Directory: D:/myfolder/coldbox-legacy-app-demo/HTMLHelper/models 
```

The `HTMLHelper/models` folders are located under `coldbox/system/modules`. Somehow this is not recognized by ACF. Do I have to modify a mapping?

**Comment by bdw429s** [comment]
@bardware ColdFusion 2018 has some serious regressions that prevent ColdBox from even working on it.  I would stay away from Adobe CF 2018 for now.  Alternatively, you can probably make it work if you manually add a `this.mappings` into your Application.crfc that points to the core HTMLHelper module.  I'm hoping Adobe will finally fix this bug in update 2.

**Comment by apendrag0n3** [comment]
Here it is - January 2019 - and I am still having this same issue with ACF 2018 and ColdBox - any hints on a fix (from either Ortis or Adobe)?

**Comment by apendrag0n3** [comment]
Brad - your previous  comment also stated "ColdFusion 2018 has some serious regressions that prevent ColdBox from even working on it" - so why would I implement a workaround for a symptom of a larger - seemingly systemic - problem?  

I found the same "symptom" - two months after this was originally posted.  

I was hoping for something more than "it's all Adobe's fault but here's a workaround - but these two won't work together anyway".

My mistake.  I won't hold expectations like that again.

**Comment by bdw429s** [comment]
>  so why would I implement a workaround for a symptom of a larger - seemingly systemic - problem?

Because you insist on using ACF 2018?  I dunno, you tell me.  We've had to basically avoid 2018 with all our clients due to these issues.  If you want to use it and workaround the issue then be my guest.  I'm not sure what else you'd like me to tell you.  This bugs are very frustrating to us because we reported them to Adobe during the beta phase and they weren't fixed in the final release nor the first updater.  We don't get to control when Adobe fixes their bugs.  We're just stuck here dealing with them just like you.   

>  "it's all Adobe's fault but here's a workaround - but these two won't work together anyway".

I think you're overstating this a bit  There _are_ serious bugs that may or may not affect you depending on what you're doing with ColdFusion.  The 3 biggest ones I'm aware of in 2018 are:
1) Auto-adding of CF Mappings broken.  This will affect any ColdBox user using modules.  The workaround is to manually add a CF mapping for every module you install (or that ships with ColdBox)
2) Mocking methods on a CFC that implements an interface now throws exceptions at runtime.  This would affect you if you're using TestBox and MockBox for method mocking.  No workaround known at this time.
3) Tons of ORM bugs, I'd have to look them all up.  This would affect anyone using the `cborm` module for sure and possibly anyone using CF's ORM in general.  

Those 3 issues hit pretty much all of our apps and many of our client's apps square in the jaw, but you can decide whether your code will be affected by them.  Like I said originally, I'd just avoid 2018 for now until Adobe tightens things down.  It's a well-known practice to avoid new versions of Adobe CF in production until at least the first updater when the bugs get fixed.  Updater 1 in 2018 was a security fix however so I think that's why it didn't fix much.  Updater 2 (with fixes for most of the issues above) should come out in Feb.  We're testing a beta of it now.

**Comment by apendrag0n3** [comment]
I "insist" on using 2018 - because that is what is deploy throughout our corporate environment - so not really as much that I "insist" on using it - as that is "all there is" in our environment. And "avoiding" it  would not be an option for my team.  

I will inform our team that ColdBox will be a "non-starter" until "at least" after 2018 Updater 2.

**Comment by bdw429s** [comment]
Yep, I get it.  You're just a cog in the wheel using what's provided.  I think your ire would be better directed at Adobe than here at me.  I didn't break anything, Adobe did.  

Here's some of the tickets in the Adobe bug tracker that we logged against 2018 so you can tell them how inconvenient these bugs are for you:

https://tracker.adobe.com/#/view/CF-4203283
https://tracker.adobe.com/#/view/CF-4203134
https://tracker.adobe.com/#/view/CF-4203133
https://tracker.adobe.com/#/view/CF-4203136
https://tracker.adobe.com/#/view/CF-4200117

**Comment by apendrag0n3** [comment]
My apologies if I seem "irate" - most definitely with the scenario and not with you. I'm in the process of upvoting those tickets.
